rust/arrow/src/compute/kernels/sort.rs

```diff
@@ -1224,6 +1460,18 @@ mod tests {
                 descending: false,
                 nulls_first: true,
             }),
+            None,
+            vec![Some(1.0), Some(f64::NAN), Some(f64::NAN), Some(f64::NAN)],
+        );
+
+        // limit
+        test_sort_primitive_arrays::<Float64Type>(
+            vec![Some(f64::NAN), Some(f64::NAN), Some(f64::NAN), Some(1.0)],
```

Comment:
I recommend using actual (values here (not just `Nan`)) as well as picking a limit other than 4 (which is the length of the array)